ucsb-cs156-f24/proj-dining-f24-16
# Implement User-Specific Review Retrieval Functionality

MERGE AFTER #52 

**Description:**
This pull request adds an API endpoint that allows users to retrieve reviews they have submitted. This ensures that users can access only their own reviews, maintaining privacy and security.

**Key Changes:**
- Test Method: Added a_logged_in_user_can_get_own_reviews_list() to verify that the endpoint correctly returns a user's reviews.
- API Endpoint: Implemented GET /api/reviews/userReviews, accessible only to authenticated users with 'ROLE_USER'.
- Repository Method: Included a method in ReviewRepository to fetch reviews by the student ID derived from the authenticated user.

**Testing:**
- Tests cover scenarios including retrieving single and multiple reviews ensuring both security and functionality are maintained.
- Jacoco is at 100 %
- mutation testing at 100 %

[Dokku instance](https://proj-dining-dev-jayster7101.dokku-16.cs.ucsb.edu/) *Read below first*
To test on dokku: 
1) must first ensure data is populated in menuItems db:
- navigate to the swagger endpoint: https://proj-dining-dev-jayster7101.dokku-16.cs.ucsb.edu/swagger-ui/index.html#/UCSBDiningMenuItems/get_menu_items
- Then enter in values ex: date/time = 2024-11-12T08:08:08 | dining-commons-code = portola | meal-code = dinner

2) Make a review : https://proj-dining-dev-jayster7101.dokku-16.cs.ucsb.edu/swagger-ui/index.html#/Review/postReview

3) Request reviews made by you only: https://proj-dining-dev-jayster7101.dokku-16.cs.ucsb.edu/swagger-ui/index.html#/Review/get_all_review_by_user_id

closes #8
